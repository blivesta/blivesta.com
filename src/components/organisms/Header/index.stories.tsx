import React from 'react';
// import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Header from '.';

storiesOf(`organisms|Header`, module).add(`default`, () => (
  <Header siteTitle="blivesta" description="description" pageLocation="/" />
));
