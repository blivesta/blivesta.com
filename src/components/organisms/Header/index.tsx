import * as React from 'react';
import { Link } from 'gatsby';
import styled from 'styled-components';

import Logo from '../../atoms/Logo';
import ContentBlock from '../../molecules/ContentBlock';

const Wrapper = styled.header``;

const H1 = styled.h1`
  font-family: ${props => props.theme.fonts.avantGarde};
  font-weight: 700;
  font-size: ${props => props.theme.fontSizes[7]};
  margin-top: -8px;
  margin-bottom: ${props => props.theme.spaces[1]};
  line-height: 1;

  ${props => props.theme.media.sm`
    font-size: ${props.theme.fontSizes[8]};
  `}
`;

const P = styled.p`
  font-family: ${props => props.theme.fonts.en};
  font-size: ${props => props.theme.fontSizes[1]};
  color: ${props => props.theme.colors.baseText[2]};
  margin-bottom: 0;

  ${props => props.theme.media.sm`
    font-size: ${props.theme.fontSizes[3]};
  `}
`;

interface HeaderProps {
  siteTitle: string;
  description: string;
}

const Header = ({ siteTitle, description }: HeaderProps) => {
  return (
    <Wrapper>
      <ContentBlock
        logo={
          <Link to="/">
            <Logo />
          </Link>
        }
      >
        <H1>{siteTitle}</H1>
        <P>{description}</P>
      </ContentBlock>
    </Wrapper>
  );
};

Header.defaultProps = {
  siteTitle: ``,
};

export default Header;
