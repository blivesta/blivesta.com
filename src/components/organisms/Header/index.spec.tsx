import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '.';

describe('Header', () => {
  it('snapshot', () => {
    const tree = shallow(<Header siteTitle="blivesta" pageLocation="/" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
