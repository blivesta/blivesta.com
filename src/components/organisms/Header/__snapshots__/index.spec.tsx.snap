// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<styled.header>
  <Styled(styled.div)
    pageLocation="/"
    siteTitle="blivesta"
  >
    <styled.nav>
      <styled.ul>
        <styled.li>
          <ForwardRef
            title="contact"
            to="/contact/"
          >
            <[object Object] />
          </ForwardRef>
        </styled.li>
        <styled.li>
          <styled.div
            aria-label="theme switch button"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <[object Object] />
          </styled.div>
        </styled.li>
      </styled.ul>
    </styled.nav>
  </Styled(styled.div)>
</styled.header>
`;
